File: formation-service/src/main/java/com/example/formationservice/models/FormationRequest.java
package com.example.formationservice.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data @NoArgsConstructor @AllArgsConstructor
public class FormationRequest {
    private Long id;
    private String name;
    private String objectif;
    private String duree;
    private Date formationDate;
}

File: rh-front/src/app/models/formationRequest.ts

export class FormationRequest {
    id: number;
    name: string;
    objectif: string;
    duree: string;
    formationDate: Date;
    // collaborateurs: Set<Collaborateur>;
    // plan: Set<Plan>;
    // demandes: Array<Demande>;

    constructor() {
        this.id = 0;
        this.name = '';
        this.objectif = '';
        this.duree = '';
        this.formationDate = new Date();
        // this.collaborateurs = new Set<Collaborateur>();
        // this.plan = new Set<Plan>();
        // this.demandes = new Array<Demande>();
    }
}